src/build/app/mod.rs

```diff
@@ -605,47 +605,20 @@ impl<'help> App<'help> {
         if self.settings.is_set(AppSettings::Multicall) {
             if let Some((argv0, _)) = it.next() {
                 let argv0 = Path::new(&argv0);
-                if let Some(command) = argv0.file_name().and_then(|f| f.to_str()) {
+                if let Some(command) = argv0.file_stem().and_then(|f| f.to_str()) {
```

Comment:
Is this fixing https://github.com/clap-rs/clap/issues/3074?

Reply:
And with that fixed, can we start using the `.md` files to actually test this? (see https://docs.rs/trycmd/)

Reply:
> Is this fixing #3074?
It should do, the improvements to the test suite since the original version showed me I hadn't considered file extensions.
I'll update the ticket metadata.
> And with that fixed, can we start using the `.md` files to actually test this? (see https://docs.rs/trycmd/)
That's neat, I'll see what I can do.

Reply:
I'm not sure how to drive trycmd.
I think removing the `ignore` from the markdown code blocks _should_ make it run the commands, but after doing that and running `cargo test --features "wrap_help yaml regex unstable-replace unstable-multicall" --test examples` still skips it.

Reply:
Looks like I had a typo.  We have to manually forward the feature flags on.  Rebase on top of #3104

Reply:
Hmm, it's expecting to have .exe on the end of the name in the output on Windows. This is going to require more thought if that's desirable.
```
Testing examples\multicall_busybox.md:27 ... failed
Exit: 2

--- stdout	expected
+++ stdout	actual
@@ -4 +4 @@
-    busybox[EXE] [OPTIONS] [APPLET]
+    busybox [OPTIONS] [APPLET]

stderr:
```

Reply:
If the help output is meant to include .exe, then I need to revert the `.file_stem` change and try something else.
I tried modifying examples to add `std::env::consts::EXE_SUFFIX` to the subcommand names, but this got unwieldy quickly.
Adding some logic to `.setting` and `.subcommand` works but is… messy, makes handling matches harder, and probably indicates the matches logic should be instead loosened if Multicall is set, but I'm going to have to familiarise myself with that code and pick this up again tomorrow.

Reply:
I tried altering aliases_to and find_subcommand, but that would change the API more extensively.
Changing the `Parser::possible_subcommand` works. It doesn't cover flag subcommands, but those aren't applet-like.
However it doesn't fix the output, because the name of the subcommand is re-output into the help rather than the string which matched it, so it _has_ to be the subcommand name, which limits reusability of subparsers under windows.
I'll have another go at explicitly adding the EXE_SUFFIX to subcommand names.

Reply:
Isn't too bad with a couple of helper functions.
I expect clap derive to have more trouble since the names are provided at compile time, but it can be made to work with a build.rs exporting an environment variable.

Reply:
Sorry, I haven't gotten to this sooner.  `[EXE]` is a variable that, on windows equals `.exe` and evaluates to nothing on Linux.  You can see we put it in our other md files to make Windows happy.

Reply:
Sorry, I was only responding to the trycmd test which it looks like you resolved.

How come you got rid of the suffix stripping in `App`?  It seems like an unnecessary burden (and a trap for people developing on Linux for both platforms) to require the developer to deal with the `.exe` suffix.

Reply:
Ah, I missed 
> If the help output is meant to include .exe, then I need to revert the .file_stem change and try something else.

I think in your case, we can strip it.  I'm actually [proposing we strip it generally](https://github.com/clap-rs/clap/issues/992#issuecomment-991200544)

Reply:
Latest version strips it.